client/dom/test/genesearch.unit.spec.ts

```diff
@@ -0,0 +1,116 @@
+import tape from 'tape'
+import * as d3s from 'd3-selection'
+import { addGeneSearchbox, string2variant } from '../genesearch.ts'
+import { hg38 } from '../../test/testdata/genomes'
+import { Menu } from '../menu'
+import { detectOne, detectGte } from '../../test/test.helpers'
+
+/* Tests
+    - Default gene search box
+    - string2variant()
+*/
+
+/**************
+ helper functions
+***************/
+
+function getHolder() {
+	return d3s.select('body').append('div').style('padding', '5px').style('margin', '5px')
+}
+
+function getRow(holder) {
+	return holder.append('div').style('border', '1px solid #aaa').style('padding', '5px')
+}
+
+function getSearchBox(holder, opts = {}) {
+	const _opts = {
+		genome: hg38,
+		tip: new Menu({ padding: '' }),
+		row: getRow(holder)
+	}
+
+	const args = Object.assign(_opts, opts)
+	addGeneSearchbox(args)
+}
+
+/**************
+ test sections
+***************/
+
+tape('\n', test => {
+	test.pass('-***- dom/genesearch -***-')
+	test.end()
+})
+
+tape('Default gene search box', async test => {
+	test.timeoutAfter(2000)
+	const holder = getHolder()
+	const tip = new Menu({ padding: '' })
+	getSearchBox(holder, { tip })
+	const searchInput = holder.select('input').node() as HTMLInputElement
+
+	// The line below is for typescript to stop complaining
+	if (!searchInput) test.fail('No gene search box created')
+	test.ok(searchInput.tagName == 'INPUT', 'Should create an input element')
+	test.equal(searchInput.placeholder, 'Gene, position, dbSNP', 'Should display the default placeholder text')
+
+	// //Check if the search box is functional
+	// await detectOne({
+	// 	target: tip.dnode,
+	// 	trigger() {
+	// 		searchInput.value = 'BRCA1'
+	// 		searchInput.dispatchEvent(new Event('keyup'))
+	// 	}
+	// })
+	// /** Leave this
+	//  * Options populate slightly slower than the menu
+	//  * Need to wait for options to appear before testing
+	//  */
+	// const options = await detectGte({
+	// 	selector: '.sja_menuoption',
+	// 	target: tip.dnode,
+	// 	count: 2
+	// })
+	// const foundGenes = options.filter((d: HTMLElement) => d.textContent?.includes('BRCA1'))
+	// test.equal(options.length, foundGenes.length, 'Should display matching gene options')
+
+	if (test['_ok']) {
+		if (tip.dnode) tip.dnode.remove()
+		holder.remove()
+	}
+	test.end()
+})
+
+tape('string2variant()', async test => {
```

Comment:
worth to add a note that cannot use delins/del variants in unit test as string2variant() will need to query a server to retrieve data